contributors/devel/sig-instrumentation/logging.md

```diff
@@ -1,37 +1,78 @@
-## Logging Conventions
+## Logging
 
-The following conventions for the klog levels to use.
-[klog](http://godoc.org/github.com/kubernetes/klog) is globally preferred to
-[log](http://golang.org/pkg/log/) for better runtime control.
+This document provides an overview of  the recommended way to develop and implement
+logging for components of Kubernetes.
+For [k/kubernetes](https://github.com/kubernetes/kubernetes) these conventions should be seen
+as a strong recommendation, however any Kubernetes or external project is welcome to follow
+the guidelines below. Most of the ideas can be applied more broadly.
 
-Shared libraries, such as `client-go`, should not use `klog.ErrorS()` and `klog.InfoS()`,
+### Logging in Kubernetes
+
+Kubernetes project uses [klog](https://github.com/kubernetes/klog) for logging.
+This library was created as a permanent fork of glog, as the original project was no longer developed. 
+However, should not be used by external projects as it is mostly developed to maintain backward compatibility.
+With the introduction of [Structured Logging](https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging), Kubernetes is being migrated to use [logr](https://github.com/go-logr/logr) as its logging interface. Any changes in klog are done to ensure a smooth transition to `logr`.
+``
+
+For projects that want to integrate with Kubernetes or write same log format we recommend using [klogr](https://github.com/kubernetes/klog/tree/main/klogr), reimplementation of klog format that exposes [logr](https://github.com/go-logr/logr) interface.
+
+### Logging Conventions
+
+### When not to log?
+
+Shared libraries, such as [`client-go`](https://github.com/kubernetes/client-go), should _not_
+write logs themselves,
 but just return `error`, because client libraries may be used in CLI UIs that wish to control output.
 
-Please see the [Structured Logging Guide](migration-to-structured-logging.md#structured-logging-in-kubernetes) for more information on how to set keys and values in structured logs.
+### How to log
 
-* klog.ErrorS() - Errors should be used to indicate unexpected behaviours in code, like unexpected errors returned by subroutine function calls.
-Logs generated by `ErrorS` command may be enhanced with additional debug information (by logging library). Calling `ErrorS` with `nil` as error may be acceptable if there is error condition that deserves a stack trace at this origin point.
+There are two main klog methods for writing logs: `klog.InfoS` and `klog.ErrorS`. Both methods are part of a [logr](https://github.com/go-logr/logr) compatible interface.
```

Comment:
I will have to update this section to cover the replacement of those calls with logging through `Logger`. This goes beyond a simple PR review, so let's get this merged first even if it is not fully up-to-date.

Reply:
sg